python_modules/dagster/dagster/_core/storage/fs_io_manager.py

```diff
@@ -199,6 +199,24 @@ def handle_output(self, context, obj):
 
     def load_input(self, context):
         """Unpickle the file and Load it to a data object."""
+        if context.has_asset_partitions:
```

Comment:
For the single-partition case, this would be a behavior change, right?

If we were building this from the ground up, I could see the wisdom in handling partitions in a unified way, independent of whether there's a single one or plural.

However, given our compatibility guarantees, I think we need to maintain the existing behavior for single partitions.

Reply:
One more thing: I noticed that our current I/O manager has special handling for `if context.dagster_type.typing_type == type(None):`. I think we'd want to retain that even in the case of multiple asset partitions.

Reply:
I agree with both comments, I added a unit test to check the backwards compatible use case and I reinstated the dagster_type check